modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala

```diff
@@ -91,13 +99,33 @@ object GlobalSnapshotConsensusFunctions {
         ordinal = lastGS.ordinal.next
         maybeTipTrigger = tipTriggersEvents.find(_.height === heightLimit)
 
+        lastTxRefs <- transactionsInRange
+          .groupBy(_.source)
+          .toList
+          .traverse {
+            case (address, txs) =>
+              TransactionReference.of(txs.maxBy(_.ordinal)).map((address, _))
```

Comment:
Please make sure that `maxBy` does not throw for empty list. Scala's fn was throwing errors while cats' fn worked on Option. just double check

Reply:
That's why I mentioned that `Set[Signed[Transaction]]` should be probably `NonEmptySet` because we can't have empty blocks. I will doublecheck that.

Reply:
Actually it never throws because the list comes from `groupBy`.

So you either have at least 1 transaction for each `address` or you have just empty `Map`